extension/jaegerremotesampling/internal/remote_strategy_cache_test.go

```diff
@@ -0,0 +1,247 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+package internal
+
+import (
+	"fmt"
+	"sync"
+	"testing"
+	"time"
+
+	"github.com/fortytw2/leaktest"
+	"github.com/jaegertracing/jaeger/thrift-gen/sampling"
+	"github.com/stretchr/testify/assert"
+)
+
+const cacheTestItemTTL = 50 * time.Millisecond
+
+var testStrategyResponseA = &sampling.SamplingStrategyResponse{
+	ProbabilisticSampling: &sampling.ProbabilisticSamplingStrategy{
+		SamplingRate: 0.1337,
+	},
+}
+
+var testStrategyResponseB = &sampling.SamplingStrategyResponse{
+	ProbabilisticSampling: &sampling.ProbabilisticSamplingStrategy{
+		SamplingRate: 0.1337,
+	},
+}
+
+func Test_serviceStrategyCache_ReadWriteSequence(t *testing.T) {
+	cache := newServiceStrategyCache(cacheTestItemTTL).(*serviceStrategyTTLCache)
+	defer func() {
+		assert.NoError(t, cache.close())
+	}()
+
+	// override clock with a test-controlled one
+	testTime := time.Unix(1690221207, 0)
+	cache.clockFn = func() time.Time {
+		return testTime
+	}
+
+	// initial read returns nothing
+	result, ok := cache.get("fooSvc")
+	assert.False(t, ok)
+	assert.Nil(t, result)
+	result, ok = cache.get("barSvc")
+	assert.False(t, ok)
+	assert.Nil(t, result)
+
+	// perform a write for fooSvc at testTime
+	firstWriteTime := testTime
+	cache.put("fooSvc", testStrategyResponseA)
+
+	// whitebox assert for internal timestamp tracking (we don't want a caching bug manifesting as stale data serving)
+	// (post-write)
+	assert.Equal(t, serviceStrategyCacheEntry{
+		retrievedAt:      testTime,
```

Comment:
```suggestion
		retrievedAt:      firstWriteTime,
```

So that this assertion will be the same as the one performed under "reading does not mutate internal cache state"?

Reply:
good call for clarity for readers